charts/kubedb-crds/templates/elasticsearch-metrics-config.yaml

```diff
@@ -0,0 +1,645 @@
+      type: gauge
+      field:
+        path: .spec.topology.data.maxUnavailable
+        type: Integer
+      metricValue:
+        valueFromPath: .spec.topology.data.maxUnavailable
+    
+      ##### Metrics for DataContent topology #####
+
+    - name: kubedb_elasticsearch_datacontent_node_replicas
+      help: "Number of replicas for elasticsearch dataContent node"
+      type: gauge
+      field:
+        path: .spec.topology.dataContent.replicas
+        type: Integer
+      metricValue:
+        valueFromPath: .spec.topology.dataContent.replicas
+    
+    - name: kubedb_elasticsearch_datacontent_node_storage_bytes
+      help: "Available storage for elastic search dataContent node"
+      type: gauge
+      labels:
+        - key: unit
+          value: bytes
+      params:
+        - key: storage
+          valuePath: .spec.topology.dataContent.storage.resources.requests.storage
+      metricValue:
+        valueFromExpression: calculateMemory(storage)
+      
+    - name: kubedb_elasticsearch_datacontent_node_info
+      help: "Elasticsearch dataContent node information"
+      type: gauge
+      labels:
+        - key: cpu_request
+          valuePath: .spec.topology.dataContent.resources.requests.cpu
+        - key: cpu_limit
+          valuePath: .spec.topology.dataContent.resources.limits.cpu
+        - key: memory_request
+          valuePath: .spec.topology.dataContent.resources.requests.memory
+        - key: memory_limit
+          valuePath: .spec.topology.dataContent.resources.limits.memory
+        - key: storage_className
+          valuePath: .spec.topology.dataContent.storage.storageClassName
+      metricValue:
+        value: 1
+    
+    - name: kubedb_elasticsearch_datacontent_node_max_unavailable
+      help: "Maximum number of pods unavailable after an eviction"
+      type: gauge
+      field:
+        path: .spec.topology.dataContent.maxUnavailable
+        type: Integer
+      metricValue:
+        valueFromPath: .spec.topology.dataContent.maxUnavailable
+
+    ##### Metrics for DataHot topology #####
+
+    - name: kubedb_elasticsearch_datahot_node_replicas
+      help: "Number of replicas for elasticsearch dataHot node"
+      type: gauge
+      field:
+        path: .spec.topology.dataHot.replicas
+        type: Integer
+      metricValue:
+        valueFromPath: .spec.topology.dataHot.replicas
+    
+    - name: kubedb_elasticsearch_datahot_node_storage_bytes
+      help: "Available storage for elastic search dataHot node"
+      type: gauge
+      labels:
+        - key: unit
+          value: bytes
+      params:
+        - key: storage
+          valuePath: .spec.topology.dataHot.storage.resources.requests.storage
+      metricValue:
+        valueFromExpression: calculateMemory(storage)
+      
+    - name: kubedb_elasticsearch_datahot_node_info
+      help: "Elasticsearch dataHot node information"
+      type: gauge
+      labels:
+        - key: cpu_request
```

Comment:
Can we have metrics like below?
- `kubedb_elasticsearch_datahot_node_memory_request`
- `kubedb_elasticsearch_datahot_node_memory_limit`
-  same for cpu.

Reply:
Sure, working on that

Reply:
Resolved